res/controllers/Reloop-Digital-Jockey-2-IE.scripts.js

```diff
@@ -0,0 +1,876 @@
+                engine.setValue("[Library]", "MoveRight", 1);
+                break;
+            case this.scratchMidiCtrl:
+            case this.scratchMidiCtrl + RDJ2.SHIFT_OFFSET:
+                engine.setValue("[Library]", "MoveLeft", 1);
+                break;
+            case this.fxDryWetMidiCtrl:
+            case this.fxDryWetMidiCtrl + RDJ2.SHIFT_OFFSET:
+                //'MoveFocusForward' is equivalent to pressing TAB key on the keyboard
+                engine.setValue("[Library]", "MoveFocusForward", 1);
+                break;
+                /* The below code will probably be removed.
+                   It allowed absolute referencing to buttons to enable
+                   moving focus backward/forward. It seems that using only
+                   'MoveFocusForward' is enough.
+
+                case RDJ2.BUTTONMAP_CH0_CH1.fxdrywet[0]:
+                case RDJ2.BUTTONMAP_CH0_CH1.fxdrywet[0] + RDJ2.SHIFT_OFFSET:
+                    engine.setValue('[Library]', 'MoveFocusBackward', 1);
+                    break;
+                case RDJ2.BUTTONMAP_CH0_CH1.fxdrywet[1]:
+                case RDJ2.BUTTONMAP_CH0_CH1.fxdrywet[1] + RDJ2.SHIFT_OFFSET:
+                    engine.setValue('[Library]', 'MoveFocusForward', 1);
+                    break;*/
+            default:
+                RDJ2.logError("Unexpected MIDI ctrl value: " + control);
+            }
+        }
+    },
+    getJogMode: function() {
+        return this.jogMode;
+    },
+    setTraxMode: function(isTraxModeEnabled) {
+        if (isTraxModeEnabled) {
+            if (this.jogMode !== RDJ2.JOGMODES.trax) {
+                this.lastNonTraxJogMode = this.jogMode;
+                this.jogMode = RDJ2.JOGMODES.trax;
+                this.input = this.inputTrax;
+                //set all LEDs on to indicate trax mode
+                if (midi.sendShortMsg) {
+                    midi.sendShortMsg(0x90, this.searchMidiCtrl, RDJ2.MIDI_ON);
+                    midi.sendShortMsg(0x90, this.scratchMidiCtrl, RDJ2.MIDI_ON);
+                    midi.sendShortMsg(0x90, this.fxDryWetMidiCtrl, RDJ2.MIDI_ON);
+                }
+            }
+        } else {
+            this.jogMode = this.lastNonTraxJogMode;
+            this.input = this.inputNormal;
+            this.updateControls();
+        }
+    },
+    unshift: function() {
+        var isLibraryModeEnabled = engine.getValue("[Master]", "maximize_library");
+        if (!isLibraryModeEnabled) {
+            this.setTraxMode(false);
+        }
+    },
+    shift: function() {
+        var isLibraryModeEnabled = engine.getValue("[Master]", "maximize_library");
+        if (!isLibraryModeEnabled) {
+            this.setTraxMode(true);
+        }
+    },
+});
+
+/* Jog Wheel */
+
+RDJ2.Deck.prototype.onJogTouch = function(channel, control, value) {
+    var currentJogMode =  this.jogModeSelector.getJogMode();
+    this.jogTouchState = RDJ2.isButtonPressed(value);
+
```

Comment:
why is the jog touch handler part of the Deck instead of the JogWheel component above?

Reply:
JogModeSelector is instantiated inside Deck component (see Deck constructor) and is responsible for controlling only the jog mode. It is a subcomponent of the Deck. The Deck itself handles the touching/spinning behaviour of the jog wheel.